test/test_ops.py

```diff
@@ -165,6 +166,34 @@ def test_numpy_ref(self, device, dtype, op):
         finally:
             torch.set_default_dtype(cur_default)
 
+    # Tests that the cpu and gpu results are consistent
+    @onlyCUDA
+    @suppress_warnings
+    @ops(_ops_and_refs_with_no_numpy_ref, dtypes=OpDTypes.any_one)
+    def test_compare_cpu(self, device, dtype, op):
+
+        if dtype not in op.dtypes:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+        if dtype not in op.dtypesIfCUDA:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+
+        def to_cpu(arg):
+            if isinstance(arg, torch.Tensor):
+                return arg.to(device='cpu')
+            return arg
+
+        samples = op.sample_inputs(device, dtype)
```

Comment:
Actually, here we could use the `reference_inputs`. WDYT @mruberry?

Reply:
Great catch, @lezcano. I agree that would be more consistent with similar tests

Reply:
added in https://github.com/pytorch/pytorch/pull/86462